clang/include/clang/CIR/Dialect/IR/CIRStdOps.td

```diff
@@ -0,0 +1,56 @@
+#ifndef MLIR_CIR_TRANSFORMS_STD_FUNCS
+#define MLIR_CIR_TRANSFORMS_STD_FUNCS
+
+include "mlir/IR/EnumAttr.td"
+
+class CIRStdOp<string functionName, dag args, dag res, list<Trait> traits = []>:
+    CIR_Op<"std." # functionName, traits> {
+  string funcName = functionName;
+  
+  let arguments = !con((ins FlatSymbolRefAttr:$original_fn), args);
+
+  let summary = "std::" # functionName # "()";
+  let results = res;
+
+  let extraClassDeclaration = [{
+    static constexpr unsigned getNumArgs() {
+      return }] # !size(args) # [{;
+    }
+    static llvm::StringRef getFunctionName() {
+      return "}] # functionName # [{";
+    }
+  }];
+
+  string argsAssemblyFormat = !interleave(
+    !foreach(
+      name,
+      !foreach(i, !range(!size(args)), !getdagname(args, i)),
+      !strconcat("$", name, " `:` type($", name, ")")
+    ), " `,` "
+  );
+
+  string resultAssemblyFormat = !if(
+    !empty(res),
+    "",
+    " `->` type($" # !getdagname(res, 0) # ")"
+  );
+
+  let assemblyFormat = !strconcat("`(` ", argsAssemblyFormat,
+                                  " `,` $original_fn `)`", resultAssemblyFormat,
+                                  " attr-dict");
+
+  let hasVerifier = 0;
+}
+
+def StdFindOp : CIRStdOp<"find",
+  (ins CIR_AnyType:$begin, CIR_AnyType:$end, CIR_AnyType:$pattern),
```

Comment:
Let's keep alignment with the original argument names: `$first`, `$last`, and `$pattern`.